### DISCUSSION TITLE
Transform after creation?

### DISCUSSION THREAD

**Discussion by tilmanb** [cduck/drawSvg#120] [Q&A] [answered]
Hi,

i'd like to be able to apply a transform to a rectangle after creation.

My usecase is that I have a 100x100 canvas and want to draw a small symbol at 12 defined positions on the canvas.

So i want to do this:

```
import drawsvg as draw

if __name__=="__main__":    

    d = draw.Drawing(100,100,origin='center')
    g = (-5,-5,5,5)

    rect_raw = draw.Rectangle(g[0],g[1],g[2],g[3],stroke='black',fill='none',stroke_width=2 )

    positions = (
        (16,9),
        (84,9),
        (49,16),
        (23,37),
        # more positions...
    )
    for x,y in positions:
        x = x - 50
        y = y - 50

        # this does nothing...
        rect_raw.transform="translate(%d,%d)" % (x,y)

        d.append(rect_raw)

    d.save_svg("c:/temp/out.svg")
```

The Drawing is created from a text file that defines the small symbol. It can be a line, circle, rectangle.

The text file reads (partly) like this:

```
Symbol1:
Line=-11,3,11,3
FullCircle=-5,0,3
FullCircle=5,0,3
```
which means "create a line and two filled circles on top".
This small symbol needs to be repeated at all of the positions in my `positions` list, so that there are 4 (in the example code) or 12 (in my final usecase) instances of the line-circle-circle symbol on the 100x100 canvas.

How can I do this?

**Comment by cduck** [comment, marked as answer]
Your code is very close to working.  What you want is the [`Use` element](https://developer.mozilla.org/en-US/docs/Web/SVG/Element/use) which draws a duplicate of the given shape at a specific x, y offset: `d.append(draw.Use(my_shape, x_offset, y_offset))`

```python
import drawsvg as draw

d = draw.Drawing(100, 100, origin='center')
g = (-5, -5, 5, 5)

rect_raw = draw.Rectangle(g[0], g[1], g[2], g[3],
                          stroke='black', fill='none', stroke_width=2)

positions = (
    (16, 9),
    (84, 9),
    (49, 16),
    (23, 37),
    # more positions...
)
for x, y in positions:
    x = x - 50
    y = y - 50

    d.append(draw.Use(rect_raw, x, y))  # <-- This line

d
```
![image](https://github.com/cduck/drawsvg/assets/2476062/c98fa4e7-d162-4af5-b35d-470859ff5637)

-----

To explain why your code didn't work, `rect_raw.transform` is not where the SVG attribute is stored.  You can use `rect_raw.args['transform']` if you need to access it.

Also, notice that every loop will modify the transform of the same rectangle object.  Instead of drawing four rectangles at different locations, the code will draw the same rectangle four times at the last position (23, 37).
